### PR TITLE
Only trigger ncov builds with new data 

### PR DESCRIPTION
**Pull request by joverlee521**
### Description of proposed changes

We do not want to waste time and resources to run rebuilds when the
ncov-ingest outputs have not changed. This commit updates the pipeline
to check the outputs from the uploads of the metadata TSV and sequences
FASTA files to verify they have been updated on S3. If the output
includes the message "files are identical" then we skip the trigger
for rebuilds.

Prompted by internal conversation on [Slack](https://bedfordlab.slack.com/archives/C01LCTT7JNN/p1669678401065859). 

### Testing

- [x] Modified trigger-on-new-data script worked as expected locally
- [x] Checks pass

### CODE REVIEW

**Comment by joverlee521** [comment]
Merging pre-review since a similar trigger has been working fine in the monkeypox ingest pipeline. 
We can always trigger builds manually if something goes wrong.

File: workflow/snakemake_rules/trigger.smk [hunk 1]
```diff
@@ -14,14 +14,19 @@ These output files are empty flag files to force Snakemake to run the trigger ru
 rule trigger_rebuild_pipeline:
     message: "Triggering nextstrain/ncov rebuild action (via repository dispatch)"
     input:
-        f"data/{database}/upload.done"
+        metadata_upload = f"data/{database}/metadata.tsv.gz.upload",
```
Comment (corneliusroemer) [inline workflow/snakemake_rules/trigger.smk:17]: ~metadata_upload = f"data/{database}/metadata.tsv.zst.upload",~

~Let's use zst compressed upload as it's:
a) the future
b) done faster so we gain an hour or so~

Comment (corneliusroemer) [reply workflow/snakemake_rules/trigger.smk:17]: Ignore what I said above - since ncov currently uses the `gz` output we need that one to be ready!

**Review by corneliusroemer — APPROVED** [review]
Looks great. There's not much harm in trying this out soon so I'll merge and we just check in the next few days if it worked successfully.

Ah I see you've already merged - even better.

**Comment by emmahodcroft** [comment]
Thanks Jover, this sounds great!

**Pull request merged**